src/full/Agda/TypeChecking/Warnings.hs

```diff
@@ -104,7 +99,7 @@ warning'_ loc w = do
     , prettyWarning w
     , prettyTCM c
     ]
-  return $ TCWarning loc r w p b
+  return $ TCWarning loc r' w p (P.render p) b
```

Comment:
Consider only rendering only the actual warning `w` without the context (`c`, `r'`).  This might require ensuring that each warning actually comes with a range, otherwise different instances of the same warning will be lumped together.